network/netcore/src/transport/tcp.rs

```diff
@@ -163,6 +188,37 @@ async fn resolve_and_connect(addr: NetworkAddress) -> io::Result<TcpStream> {
     }
 }
 
+async fn connect_via_proxy(proxy_addr: String, addr: NetworkAddress) -> io::Result<TcpStream> {
+    let protos = addr.as_slice();
+
+    if let Some(((host, port), _addr_suffix)) = parse_tcp(protos) {
+        let mut s = TcpStream::connect(proxy_addr).await?;
+        let mut buffer = [0; 1024];
+
+        s.write_all(&format!("CONNECT {0}:{1} HTTP/1.1\r\n\r\n", host, port).into_bytes())
```

Comment:
Nit: I'm not a fan of 1 letter variables, because I have to scroll up until I see where it's initialized to tell what it is.